coordinator/src/routes/util.ts

```diff
@@ -18,18 +19,36 @@ export function handlePollableResponse(spineResponse: SpineDirectResponse | Spin
   }
 }
 
-type Handler<T> = (requestPayload: T, responseToolkit: Hapi.ResponseToolkit) => Hapi.ResponseObject | Promise<Hapi.ResponseObject>
+type Handler<T> = (
+  requestPayload: T, responseToolkit: Hapi.ResponseToolkit
+) => Hapi.ResponseObject | Promise<Hapi.ResponseObject>
 
-export function validatingHandler(requireSignature: boolean, handler: Handler<Bundle>) {
+export type MessageType = "Prescription" | "Cancellation"
+
+function getMessageType(messageHeader: fhir.MessageHeader): MessageType {
+  const eventCode = messageHeader.eventCoding.code
+  if (eventCode === "prescription-order") return "Prescription"
+  else if (eventCode === "prescription-order-update") return "Cancellation"
+  else throw new Error(`Unknown Message Type: ${eventCode}`)
+}
+
+export function identifyMessageType(bundle: fhir.Bundle): MessageType {
+  const messageHeader = getMessageHeader(bundle)
+  return getMessageType(messageHeader)
+}
+
+export function validatingHandler(requireSignature: boolean, handler: Handler<fhir.Bundle>) {
   return async (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit): Promise<Hapi.ResponseObject> => {
     const requestPayload = getPayload(request)
-    const validation = requestValidator.verifyPrescriptionBundle(requestPayload, requireSignature)
+    const validation = requestValidator.verifyBundle(requestPayload, requireSignature)
     if (validation.length > 0) {
       const response = toFhirError(validation)
       const statusCode = requestValidator.getStatusCode(validation)
       return responseToolkit.response(response).code(statusCode)
+    } else {
+      const validatedPayload = requestPayload as fhir.Bundle
```

Comment:
added const here to be explicit that the payload is now a `fhir.Bundle` since it has been validated